### ISSUE TITLE
Scalable OCR & Translation Quality Evaluation Framework

### ISSUE THREAD

**Issue by JDerekLomas** [Embassy-of-the-Free-Mind/sourcelibrary-v2#1329]
## Vision

Build a systematic, reproducible quality evaluation framework for OCR and translation across all scripts and languages in Source Library. The goal is a `/qa-eval` skill that produces research-grade blog posts with quantitative results, usable as the basis for an academic paper on scaling quality evaluations for historical document AI.

## Background

In the Bhutan collection session (April 21-23, 2026), we built the first prototype of this system while importing 1,325 Bhutanese manuscripts (232K pages). Key findings:

- **OCR consistency**: Claude Opus 4.6 = 100% modal consistency (3/3 identical), Gemini Flash 3 = 80% (4/5 identical at temp=0, with mode switching)
- **Cross-model agreement**: 89.4% character similarity between Opus and Flash on dbu med cursive
- **Gemini 3.1 Flash Lite hallucinated** the genre of an astrological text entirely — called it a "ritual manual for weather control"
- **Translation comparison**: AI translation of Bardo Thodol opening matches scholarly translation in meaning but diverges on terminology (dgongs pa → "wisdom" vs "intention")
- Blog post draft: `docs/blog-tibetan-ocr-benchmark.md`

### 1. OCR Consistency (Multi-Run)
- Run same page through each model N times (N=3-5) at temp=0
- **Modal Consistency Rate (MCR)**: % of runs producing majority output
- **Pairwise character similarity** (Levenshtein-based)
- **Syllable-level agreement** (tsheg-delimited for Tibetan, word-delimited for Latin scripts)
- Models: Gemini 3 Flash, Gemini 3.1 Flash Lite, Claude Opus 4.6, Claude Sonnet 4.6
- Strip XML metadata before comparison

### 2. Cross-Model Agreement
- Best-run from each model compared pairwise
- Character similarity, aligned word-group comparison
- Identify systematic divergences (which words do models disagree on?)

### 3. Comparison to Human Translation
- Align AI translation against published scholarly translations (Lotsawa House, Evans-Wentz, BDRC etexts)
- **BLEU/ROUGE scores** for translation quality
- **Terminology divergence analysis** — where AI and humans choose different terms for the same Tibetan

### 4. Comparison to Proofread Etexts (Ground Truth OCR)
- BDRC has proofread Unicode etexts for many canonical texts
- OpenPecha, Esukhia Derge Kangyur (gold standard)
- **Character Error Rate (CER)** against known-good transcriptions
- Sources: `library.bdrc.io`, `github.com/OpenPecha-Data`, `github.com/Esukhia/derge-kangyur`

### 5. Embedding-Space Evaluation (Novel)
**This is the most interesting research direction.**

Given a text that exists in:
- Original language (Tibetan/Latin/Arabic)
- Human translation (English)
- AI translation (English)

Embed all three with the same model (Gemini `embedding-2-preview`). Then measure:
- `d(original, human_translation)` — baseline semantic distance
- `d(original, ai_translation)` — AI translation distance
- `d(human_translation, ai_translation)` — direct translation comparison
- `d(ai_run1, ai_run2)` — embedding consistency across runs

**Key questions:**
- Does embedding distance correlate with translation quality?
- Can we detect hallucination via embedding divergence? (If AI "translates" an astrological text as a ritual manual, the embedding should diverge from the original)
- How does temperature affect embedding variance?
- Can we use embedding alignment as a cheap proxy for translation quality at scale (instead of BLEU against ground truth)?

### 6. Script/Language Benchmarking Matrix
Run the full evaluation suite on representative pages from each major script family:

| Script | Example | Difficulty | Status |
|---|---|---|---|
| Latin (printed) | Paracelsus 1616 | Easy | Baseline |
| Latin (manuscript) | Medieval codex | Medium | Needs eval |
| German (Fraktur) | 18th c. printed | Medium | Needs eval |
| Arabic (printed) | Kitab al-Najat | Medium | Needs eval |
| Hebrew (Rashi) | Zohar commentary | Hard | **Known failure** |
| Tibetan (dbu can) | Neyphug canonical | Medium | Evaluated |
| Tibetan (dbu med) | Gangtey cursive | Hard | Evaluated |
| Chinese (classical) | Daozang woodblock | Medium | Needs eval |
| Sanskrit (Devanagari) | Printed edition | Medium | Needs eval |
| Greek (ancient) | Aldine edition | Medium | Needs eval |

For each: MCR, cross-model agreement, CER (where ground truth exists), embedding analysis.

### 7. "New Corpus Readiness" Score
When we acquire a new linguistic corpus (e.g., the Bhutanese manuscripts), run a standardized evaluation on a sample (10-20 random pages) that produces:
- **OCR readiness**: MCR + cross-model agreement → "high/medium/low confidence"
- **Translation readiness**: Embedding alignment + terminology analysis → "publishable/draft/unusable"
- **Hallucination risk**: MCR < 67% OR genre misidentification → flag

## Output Format

The skill should produce a markdown blog post + structured JSON results:
```
/qa-eval --corpus=bhutan --sample=20 --models=flash,opus
```
→ `docs/qa-eval-bhutan-2026-04-23.md` (blog post)
→ `scripts/eval/qa-eval-bhutan-2026-04-23.json` (structured data)

## Related Work

Key papers (all cited in `docs/blog-tibetan-ocr-benchmark.md`):
- GlotOCR Bench (2026) — Tibetan <10% accuracy on general OCR models
- Wang & Wang (2025, arXiv:2503.16974) — 3-5 run aggregation improves consistency
- Lopresti & Zhou (1996) — consensus voting eliminates 20-50% of OCR errors
- Conformal Risk Control for OCR (arXiv:2603.19790) — accept/abstain with statistical guarantees
- Manchu VLM fine-tuning (arXiv:2507.06761) — 93% on real handwriting with 60K synthetic images
- OmniOCR (arXiv:2602.21042) — Dynamic LoRA for minority scripts
- BDRC OCR App (March 2025) — Easter2 architecture, 43K training samples
- "OCR Post-Correction: No Free Lunches" (arXiv:2502.01205)
- "Seeing is Believing?" (arXiv:2506.20168) — VLM hallucination in degraded OCR

## Resources
- BDRC etexts: 5M pages, 30M scanned — `library.bdrc.io`
- OpenPecha: 4K+ proofread works — `github.com/OpenPecha-Data`
- Esukhia Derge Kangyur: gold standard — `github.com/Esukhia/derge-kangyur`
- Lotsawa House: Tibetan+English side-by-side — `lotsawahouse.org`
- Monlam AI OCR: dedicated Tibetan model — `monlam.ai/model/ocr`
- Our Bhutan corpus: 1,325 EAP manuscripts, 241K pages

## Prior Work in This Repo
- Blog post: `docs/blog-tibetan-ocr-benchmark.md`
- OCR consistency test script: `_tmp-ocr-consistency.mjs`
- Test translation scripts: `_tmp-test-tibetan-ocr.mjs`
- EAP archiver: `scripts/workers/archive-eap.mjs`
- Bhutan import scripts: `_tmp-batch-import-bhutan*.mjs`, `_tmp-eap-import-*.mjs`

## Paper Potential

"Scaling Quality Evaluation for Historical Document AI: A Multi-Model, Multi-Script Framework"

Novel contributions:
1. Modal Consistency Rate as a practical metric for pipeline OCR
2. Embedding-space evaluation of translation quality without ground truth
3. Cross-model agreement as a hallucination detector
4. Standardized "corpus readiness" score for new acquisitions
5. Evaluation across 10+ historical scripts with 4 VLMs

**Comment by JDerekLomas** [comment]
## Additional: Image Pre-Processing and OCR Quality

From inspecting the published Bhutanese manuscripts (e.g., [Padmasambhava Biography p1](https://sourcelibrary.org/book/69e76132cc48e59ad74ee251/page/69e76132cc48e59ad74ee252)):

**The raw manuscript images are very dark** — the pecha folios on dark backgrounds don't magnify well and may be hurting OCR quality. We should test:

### Image Enhancement Experiments
1. **Background removal** — segment the pecha folio from the dark background
2. **Histogram equalization** — enhance contrast (we already have this for grayscale scans via `enhanced_photo` field, PR #1175)
3. **Adaptive thresholding** — binarize for cleaner text extraction
4. **White balance correction** — the EAP scans have a warm/yellow cast from aged paper

### Measure Impact on OCR
- Run the same page through OCR with: raw image, histogram-enhanced, thresholded, background-removed
- Compare MCR and cross-model agreement for each variant
- Does enhancement help dbu med (cursive) more than dbu can (formal)?

### Multi-Folio Splitting
- Each image contains 2-4 pecha folios — should we split them into individual folios before OCR?
- Gemini handles multi-folio naturally (transcribes each separately), but splitting might improve consistency
- Test: same page OCR'd as whole image vs individual cropped folios

This connects to the existing scan quality enhancement pipeline (`scripts/enhance-scan-quality.mjs`) which uses per-page histogram analysis. We could adapt it for the EAP corpus.
